File: Dot Net Code/AgroRent/Models/Payment.cs
using System.ComponentModel.DataAnnotations;

namespace AgroRent.Models
{
    public class Payment : BaseEntity
    {
        [Key]
        public int Id { get; set; }

        public string? PaymentId { get; set; }      // Razorpay payment ID

        public string? OrderId { get; set; }        // Razorpay order ID

        public PaymentStatus Status { get; set; }         // CREATED, PAID, FAILED

        [Required]
        public double Amount { get; set; }

        public DateTime Timestamp { get; set; }

        // Navigation properties
        public virtual Booking? Booking { get; set; }
    }
}

File: Springboot Code/AgroRent/src/main/java/com/majorproj/agrorent/entities/Payment.java
package com.majorproj.agrorent.entities;

import java.time.LocalDateTime;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Entity
@NoArgsConstructor
@Getter
@Setter
@Table(name="payments")
@ToString(callSuper = true,exclude = {"booking"})
public class Payment extends BaseEntity{
	@Id 
	@GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
	
    private String paymentId;      // Razorpay payment ID
    private String orderId;        // Razorpay order ID
    @Enumerated(EnumType.STRING)
    private PaymentStatus status;         // CREATED, PAID, FAILED
    @Column(nullable = false)
    private double amount;
    private LocalDateTime timestamp; 

    @OneToOne
    private Booking booking;
}
